demos/iperf3/cloud/test.py

```diff
@@ -11,121 +12,80 @@
+    os.chdir(projDir)
     
     if command =="delete":
-        deleteClustersList([mbg1, mbg2])
+        gw1.deleteCluster(runBg=True)
+        gw2.deleteCluster()
+
         exit()
     elif command =="clean":
-        cleanClustersList([mbg1, mbg2])
+        gw1.cleanCluster()
+        gw2.cleanCluster()
         exit()
 
-    #Create k8s cluster
-    createCluster(cluster=mbg1, run_in_bg=True , machineType = machineType)
-    createCluster(cluster=mbg2, run_in_bg=False, machineType = machineType)
-        
-    # #Setup MBG1
-    checkClusterIsReady(mbg1)
-    mbg1Ip=mbgBuild(mbgcPort=mbgcPort)
-    mbgSetup(mbg1, dataplane, mbg1crtFlags, gwctlName=gwctl1, mbgIp=mbg1Ip, mbgcPort=mbgcPort)
+    ### build docker environment 
+    printHeader("Build docker image")
+    os.system("make build")
+    os.system("sudo make install")
     
-    #Build MBG2
-    checkClusterIsReady(mbg2)
-    mbg2Ip=mbgBuild(mbgcPort=mbgcPort)
-    mbgSetup(mbg2, dataplane, mbg2crtFlags, gwctlName=gwctl2, mbgIp=mbg2Ip, mbgcPort=mbgcPort)
+    gw1.machineType=machineType
+    gw2.machineType=machineType
     
+    iperf3Test(gw1, gw2, testOutputFolder, args["logLevel"], args["dataplane"])
 
-    #Add MBG Peer
-    connectToCluster(mbg2)
-    gwctl2Pod =getPodName("gwctl")
-    printHeader("Add MBG1 MBG2")
-    runcmd(f'kubectl exec -i {gwctl2Pod} -- ./gwctl create peer --name {mbg1.name} --host {mbg1Ip} --port {mbgcPort}')
+    # iPerf3 test
+    gw1.useCluster()
+    podIperf3,_= getPodNameIp(srcSvc)
 
-            
-    # Send Hello
-    printHeader("Send Hello commands")
-    runcmd(f'kubectl exec -i {gwctl2Pod} -- ./gwctl hello')
-        
-    #Add services 
-    connectToCluster(mbg1)
-    gwctl1Pod =getPodName("gwctl")
-    runcmd(f"kubectl create -f {folMn}/iperf3-client/iperf3-client.yaml")
-    waitPod(srcSvc)
-    podIperf3 =getPodIp(srcSvc)
-    runcmd(f'kubectl exec -i {gwctl1Pod} -- ./gwctl create export --name {srcSvc} --host {podIperf3} --description {srcSvc}')
-    
-    connectToCluster(mbg2)
-    runcmd(f"kubectl create -f {folMn}/iperf3-server/iperf3.yaml")
-    runcmd(f"kubectl create service nodeport iperf3-server --tcp=5000:5000 --node-port=30001")
-    waitPod(destSvc)
-    destSvcIp =getPodIp(destSvc)
-    runcmd(f'kubectl exec -i {gwctl2Pod} -- ./gwctl create export --name {destSvc} --host {destSvcIp}:5000 --description {destSvc}')
-    
-    #Expose destination service
-    printHeader("\n\nStart exposing connection")
-    runcmdb(f'kubectl exec -i {gwctl2Pod} -- ./gwctl expose --service {destSvc}')
-
-    #Test MBG1
-    connectToCluster(mbg1)
-    podIperf3= getPodName(srcSvc)
-    mbgPod,mbgPodIP=getPodNameIp("mbg")
-    mbg1LocalPort, mbg1ExternalPort = getMbgPorts(mbgPod,destSvc)
     for i in range(2):
         printHeader(f"iPerf3 test {i}")
-        cmd = f'kubectl exec -i {podIperf3} --  iperf3 -c {mbgPodIP} -p {mbg1LocalPort} -t 40'
+        cmd = f'kubectl exec -i {podIperf3} --  iperf3 -c iperf3-server -p {5000} -t 40'
```

Comment:
Do we have a cleanup script to delete the clusters? or its in the Makefile?

Reply:
Not a script but the test parameter you can decide if you want to clean or delete the clusters but you need to do it manually.